src/ui/base-page-storyEdit.js

```diff
@@ -215,31 +205,51 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
                                     }
                                 }
                             }
+                        },
+                        story: {
+                            options: {
+                                model: "{storyEditor}.story.model",
+                                modelRelay: {
+                                    contentEmptyBlockFilter: {
+                                        target: "content",
+                                        singleTransform: {
+                                            type: "fluid.transforms.free",
+                                            func: "sjrk.storyTelling.base.page.storyEdit.removeEmptyBlocks",
+                                            args: ["{storyPreviewer}.story.model.content", "{storyEdit}.options.blockContentValues"]
+                                        }
+                                    }
+                                }
+                            }
                         }
                     }
                 }
             }
         }
     });
 
-    /* Removes all empty blocks from a given array of story blocks
-     * - "blocks": an array of story blocks
-     * - "blockContentValues": a collection of arrays which outline the values
+    /* Removes all empty blocks from a given collection of story blocks
+     * - "blocks": a collection of story blocks (sjrk.storyTelling.block)
+     * - "blockContentValues": a hash map of block types which outlines the values
      *      that, if at least one is truthy, means a particular block is not empty
      */
-    sjrk.storyTelling.base.page.storyEdit.removeEmptyBlocks = function (storyModel, blockContentValues) {
-        storyModel.content = fluid.remove_if(storyModel.content, function (block) {
-            return sjrk.storyTelling.base.page.storyEdit.isEmptyBlock(block, blockContentValues[block.blockType]);
+    sjrk.storyTelling.base.page.storyEdit.removeEmptyBlocks = function (blocks, blockContentValues) {
+        var filteredBlocks = [];
+
+        fluid.each(blocks, function (block) {
+            if (!sjrk.storyTelling.base.page.storyEdit.isEmptyBlock(block, blockContentValues[block.blockType])) {
+              filteredBlocks.push(block);
+            }
         });
 
-        return storyModel;
+        return filteredBlocks;
     };
 
     /* Returns true if a block is determined to be empty, based on the values
      * listed in blockContentValues. If at least one of those values is truthy,
      * the block is not empty.
-     * - "block": a story block
-     * - "blockContentValues": an array of the values as described above
+     * - "block": a single story block (sjrk.storyTelling.block)
+     * - "blockContentValues": a hash map of block types which outlines the values
+     *      that, if at least one is truthy, means a particular block is not empty
```

Comment:
I'm debating in my head if this is the right check or not. It seems that blocks would only ever hold strings. That is text or urls. Is that correct? Should other values be stripped. That is, would you accept a Number, Array, Object, etc. Also, it would accept a string only made of whitespace, or invalid URLs. In terms of users interacting through the UI, I think you'd only really have to worry about strings only made of whitespace, but it's probably still worth investigating how you handle the other conditions as well.

Reply:
That's a good question. In terms of what is currently accepted, it is the case that it's only going to be strings or URL's, though it's possible that other values could be added for different block types in the future. We had discussed a "drawing" type block at some point, so I can see that the implementation of that would potentially have more complex data stored at a particular point on its model, in which case limiting the values to strings may not go so well.

I agree that checking for all-whitespace strings is a good idea. That might be something left to field validation (currently nonexistent) rather than the filter. What do you think?

Reply:
It's good to note that at the moment an empty array or object would also pass. 

Regarding validation and filtering, in an ideal case you'd have both. This way the user knows right away that something is invalid, and we also prevent programmatic manipulation of values.